linux/utils/enable_auto_login.yml

```diff
@@ -1,84 +1,117 @@
 
-# Enable GNOME auto login for RHEL/CentOS/OracleLinux/Ubuntu/Debian
-- block:
-    # Update [daemon] section
-    - include_tasks: ../../common/update_ini_style_file.yml
-      vars:
-        file_path: "{{ gdm_conf_path }}"
-        section_name: "daemon"
-        option_name: "{{ item[0] }}"
-        option_value: "{{ item[1] }}"
-      with_list:
-        - ['AutomaticLogin', '{{ autologin_user }}']
-        - ['AutomaticLoginEnable', 'True']
+- name: "Enable autologin in display manager config file path"
+  block:
+    - name: "Check display manager config file status"
+      ansible.builtin.stat:
+        path: "{{ dm_conf_path }}"
+      register: dm_stat_result
+      delegate_to: "{{ vm_guest_ip }}"
+
+    - name: "{{ dm_conf_path }} exists. Auto login can be enabled"
+      ansible.builtin.set_fact:
+        enable_autologin: true
       when:
-        - gdm_stat_result.stat is defined
-        - gdm_stat_result.stat.exists is defined
-        - gdm_stat_result.stat.exists
-  when:
-    - guest_os_family in ["RedHat", "Debian"]
-    - enable_gnome_autologin is defined
-    - enable_gnome_autologin
+        - dm_stat_result is defined
+        - dm_stat_result.stat is defined
+        - dm_stat_result.stat.exists is defined
+        - dm_stat_result.stat.exists
+
+    - name: "Enable auto login for user {{ autologin_user }}"
+      block:
+        - name: "Enable auto login for user {{ autologin_user }} in {{ guest_os_ansible_distribution }}"
+          block:
+            - include_tasks: ../../common/update_ini_style_file.yml
+              vars:
+                file_path: "{{ dm_conf_path }}"
+                section_name: "{{ dm_conf_section }}"
+                option_name: "{{ item.key }}"
+                option_value: "{{ item.value }}"
+              with_list: "{{ dm_autologin_options | dict2items }}"
+          when: guest_os_display_manager in ["gdm", "lightdm", "fly-dm", "lxdm", "sddm"]
 
-# Enable auto login for SLES/SLED
-- block:
-    - name: "Enable GNOME autologin for {{ guest_os_ansible_distribution }}"
-      include_tasks: replace_or_add_line_in_file.yml
-      vars:
-        file: "{{ gdm_conf_path }}"
-        reg_exp: "DISPLAYMANAGER_AUTOLOGIN=.*"
-        line_content: "DISPLAYMANAGER_AUTOLOGIN={{ autologin_user }}"
-  when:
-    - guest_os_ansible_distribution in ["SLES", "SLED"]
-    - enable_gnome_autologin is defined
-    - enable_gnome_autologin
+        # Enable auto login for SLES/SLED
+        - name: "Enable auto login for user {{ autologin_user }} in {{ guest_os_ansible_distribution }}"
+          include_tasks: replace_or_add_line_in_file.yml
+          vars:
+            file: "{{ dm_conf_path }}"
+            reg_exp: "DISPLAYMANAGER_AUTOLOGIN=.*"
+            line_content: "DISPLAYMANAGER_AUTOLOGIN={{ autologin_user }}"
+          when:
+            - guest_os_display_manager == "xdm"
+            - guest_os_family == "Suse"
+      when: enable_autologin
```

Comment:
`enable_autologin` needs a default value

Reply:
Default value is set at line 10.